ast/dml.go

```diff
@@ -1473,7 +1473,66 @@ type UpdateStmt struct {
 
 // Restore implements Node interface.
 func (n *UpdateStmt) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("UPDATE ")
+
+	if n.TableHints != nil && len(n.TableHints) != 0 {
+		ctx.WritePlain("/*+ ")
+		for i, tableHint := range n.TableHints {
+			if err := tableHint.Restore(ctx); err != nil {
+				return errors.Annotatef(err, "An error occurred while restore UpdateStmt.TableHints[%d]", i)
+			}
+		}
+		ctx.WritePlain("*/ ")
+	}
+
+	switch n.Priority {
+	case mysql.LowPriority:
+		ctx.WriteKeyWord("LOW_PRIORITY ")
+	}
+	if n.IgnoreErr {
+		ctx.WriteKeyWord("IGNORE ")
+	}
+
+	if err := n.TableRefs.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occur while restore UpdateStmt.TableRefs")
+	}
+
+	ctx.WriteKeyWord(" SET ")
+	assignmentListLen := len(n.List)
+	for ix, assignment := range n.List {
+		assignment.Column.Restore(ctx)
+		ctx.WritePlain("=")
+		assignment.Expr.Restore(ctx)
+
+		if ix != assignmentListLen-1 {
+			ctx.WritePlain(", ")
+		}
+	}
+
+	if n.Where != nil {
+		ctx.WriteKeyWord(" WHERE ")
+		if err := n.Where.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occur while restore UpdateStmt.Where")
+		}
+	}
+
+	if !n.MultipleTable {
```

Comment:
It can omit the check for Order / Limit in multi-table update syntax with this condition check.

Reply:
This will result in 3 checks in single-table syntax though. Anyway these checks are practically free so I don't think it worth an extra indentation level 🙂.

Reply:
yeah, your point is reasonable.